dashboard/models.py

```diff
@@ -197,13 +199,18 @@ def json(self):
         :rtype: Dict
         """
 
-        return {
-            self.course.canvas_id: {
-                'fa': int(self.show_files_accessed),
-                'ap': int(self.show_assignment_planning),
-                'gd': int(self.show_grade_distribution),
+        try:
+            return {
+                self.course.canvas_id: {
+                    'fa': int(self.show_files_accessed),
+                    'ap': int(self.show_assignment_planning),
+                    'gd': int(self.show_grade_distribution),
+                }
             }
-        }
+        except ObjectDoesNotExist:
+            logger.warn(f"CourseViewOption does not exist in Course table, skipping")
```

Comment:
I feel the reason for exception should be printed from the Exception object as it would have additional info. isn't this should be an error than a warning?

Reply:
I tried to get additional information (like the id that was causing an issue) but I couldn't figure out how to get it after about an hour of research. If you try to access "self" it causes an Exception. It might be easier to figure this out if that debugging patch was applied. I feel like the exception might have to be re-thrown and caught at a higher level for more details.